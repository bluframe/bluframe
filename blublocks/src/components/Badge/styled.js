// @flow

import type { Palette, Theme } from "@bluframe/blublocks"
import styled, { type StyledComponent } from "styled-components"

export const Wrapper: StyledComponent<
  {| +color: $Keys<Palette> |},
  Theme,
  HTMLElement
> = styled.span`
  align-items: center;
  background-color: ${({ theme, color }): string => theme.palette[color].main};
  border-radius: 10px;
  color: ${({ theme, color }): string => theme.palette[color].contrast};
  display: flex;
  flex-wrap: wrap;
  font-size: 14px;
  justify-content: center;
  line-height: 1;
  padding: 4px 8px;
`
