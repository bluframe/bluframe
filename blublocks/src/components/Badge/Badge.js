// @flow

import type { ComponentProps } from "."
import React from "react"
import { Wrapper } from "./styled"

const Badge = ({
  className,
  color = "primary",
  label
}: ComponentProps): React$Node => (
  <Wrapper aria-label={`badge-${label}`} className={className} color={color}>
    {label}
  </Wrapper>
)

export default Badge
