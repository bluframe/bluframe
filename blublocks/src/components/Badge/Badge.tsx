import { ComponentProps } from "."
import React from "react"
import { Wrapper } from "./styled"

const Badge = ({ className, color = "primary", label }: ComponentProps) => (
  <Wrapper aria-label={`badge-${label}`} className={className} color={color}>
    {label}
  </Wrapper>
)

export default Badge
