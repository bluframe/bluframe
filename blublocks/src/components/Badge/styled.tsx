import { Palette, Theme } from "styles/theme"
import styled from "styled-components"

export const Wrapper = styled.span`
  align-items: center;
  background-color: ${({
  theme,
  color
}: {
  color: keyof Palette
  theme: Theme
}): string => theme.palette[color].main};
  border-radius: 10px;
  color: ${({ theme, color }: { color: keyof Palette; theme: Theme }): string =>
    theme.palette[color].contrast};
  display: flex;
  flex-wrap: wrap;
  font-size: 14px;
  justify-content: center;
  line-height: 1;
  padding: 4px 8px;
`
