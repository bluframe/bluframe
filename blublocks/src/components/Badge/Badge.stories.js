// @flow

import Badge, { type Props } from "components/Badge"
import React from "react"
import type { Story } from "@storybook/react"

const stories = {
  component: Badge,
  title: "Components/Badge"
}

const Template = (args: Props) => <Badge {...args} />

export const Default: Story<Props> = Template.bind({})
Default.args = {
  label: "Apples"
}

export const Secondary: Story<Props> = Template.bind({})
Secondary.args = {
  color: "secondary",
  label: "Apples"
}

export default stories
