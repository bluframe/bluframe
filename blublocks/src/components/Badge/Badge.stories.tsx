import Badge from "components/Badge"
import { StoryObj } from "@storybook/react"

const meta = {
  component: Badge,
  title: "Components/Badge"
}

type Story = StoryObj<typeof Badge>

export const Default: Story = {
  args: {
    label: "Apples"
  }
}

export const Secondary: Story = {
  args: {
    color: "secondary",
    label: "Apples"
  }
}

export default meta
