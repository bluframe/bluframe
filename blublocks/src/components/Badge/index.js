// @flow

import Badge from "./Badge"
import type { BadgeProps } from "@bluframe/blublocks"

export type Props = {|
  ...BadgeProps
|}

export type ComponentProps = {|
  ...Props
|}

export default Badge
