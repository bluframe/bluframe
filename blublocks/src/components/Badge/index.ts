import Badge from "./Badge"
import { Palette } from "styles/theme"

export interface Props {
  className?: string
  color?: keyof Palette
  label: string
}

export interface ComponentProps extends Props {}

export default Badge
