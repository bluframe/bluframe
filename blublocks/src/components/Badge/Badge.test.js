// @flow

import Badge from "./Badge"
import React from "react"
import { render } from "tests"

jest.mock(".", () => ({}))

describe("Badge", () => {
  it("renders with default props", () => {
    const { container, getByText } = render(<Badge label="Apples" />)
    expect(getByText("Apples")).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders with secondary color", () => {
    const { container } = render(<Badge color="secondary" label="Apples" />)
    expect(container.firstChild).toHaveAttribute("color", "secondary")
    expect(container.firstChild).toMatchSnapshot()
  })

  it("has expected aria-label", () => {
    const { getByLabelText } = render(<Badge label="Apples" />)
    expect(getByLabelText("badge-Apples")).toBeInTheDocument()
  })
})
