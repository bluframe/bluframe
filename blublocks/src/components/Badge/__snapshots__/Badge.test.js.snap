// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders with default props 1`] = `
<span
  aria-label="badge-Apples"
  class="styled__Wrapper-sc-1n59qnd-0 cbqPwt"
  color="primary"
>
  Apples
</span>
`;

exports[`Badge renders with secondary color 1`] = `
<span
  aria-label="badge-Apples"
  class="styled__Wrapper-sc-1n59qnd-0 cUpfLf"
  color="secondary"
>
  Apples
</span>
`;
